### PR TITLE
Fix Accept header overwritten issue when "xhr: true" in integration test

### PR DESCRIPTION
**Pull request by darkbaby123**
### Summary

Fixes https://github.com/rails/rails/issues/25859

### CODE REVIEW

File: actionpack/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+*   Fix `xhr: true` issue in `ActionDispatch::Integration::RequestHelpers`.
```
Comment (kaspth) [inline actionpack/CHANGELOG.md]: Don't think the title says much. I think we could condense this to:

```markdown
*   Don't override the `Accept` header in integration tests when called with `xhr: true`.

     Fixes #25859.

    *David Chen*
```

File: actionpack/lib/action_dispatch/testing/integration.rb [hunk 2]
```diff
@@ -360,15 +360,17 @@ def process(method, path, params: nil, headers: nil, env: nil, xhr: false, as: n
             "HTTP_ACCEPT"    => accept
           }
 
+          # # Create a wrapped_headers for key transformation
```
Comment (kaspth) [inline actionpack/lib/action_dispatch/testing/integration.rb]: This comment doesn't add much, let's just remove it.

**Comment by kaspth** [comment]
We should write regression tests for this as well to prove neither `HTTP_ACCEPT` nor `Accept` are overriden.

Can we also put that lovely investigative work you did in the issue into your commit message? That way future readers won't have to go to GitHub for info.

**Comment by darkbaby123** [comment]
Can you tell me which file the test should be written? If I should create a new test file, which file name and path do you suggest? I thought to write a test before but didn't find a proper place. Because this is a issue in test helper.

**Comment by kaspth** [comment]
@darkbaby123 somewhere in this test case class might be a good place: https://github.com/rails/rails/blob/8f26f31934b4f0c694a081d2a89996f3935768d1/actionpack/test/controller/integration_test.rb#L400

**Comment by darkbaby123** [comment]
@kaspth Yes. I just finished the work and put the test into that file. Thanks for the test code so I can find a bug for my previous solution! The changelog and commit and also modified.

File: actionpack/test/controller/integration_test.rb [hunk 3]
```diff
@@ -727,6 +728,16 @@ def test_respect_removal_of_default_headers_by_a_controller_action
     assert_includes @response.headers, "c"
   end
 
+  def test_accept_not_overriden_when_xhr_true
+    with_test_route_set do
+      get "/get", headers: { "Accept" => "application/json" }, xhr: true
```
Comment (kaspth) [inline actionpack/test/controller/integration_test.rb]: Let's add a test for `HTTP_ACCEPT` too.

**Comment by kaspth** [comment]
Thanks for the help! I've backported this to the pending 5.0.1 as well: 9047d9e

**Comment by darkbaby123** [comment]
@kaspth Glad to help!

**Pull request merged**